code/modules/buildmode/submodes/forcemove.dm

```diff
@@ -0,0 +1,78 @@
+// This tool calls forceMove() on an object/mob that is selected with right-click.
+// Once the atom is moved, its selection is cleared to avoid any further moves by accident.
+/datum/buildmode_mode/forcemove
+	key = "forcemove"
+	var/atom/movable/selected_atom		// The atom we want to move
+	var/image/selected_overlay			// Overlay for the selected atom only visible for the build mode user
+
+/datum/buildmode_mode/forcemove/show_help(mob/user)
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+	to_chat(user, "<span class='notice'>Left Mouse Button on obj/mob = Select destination</span>")
+	to_chat(user, "<span class='notice'>Right Mouse Button on obj/mob = Select atom to move</span>")
+	to_chat(user, "<span class='notice'><b>Notice:</b> You need to select the movable atom first, then left-click its destination.</span>")
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+
+/datum/buildmode_mode/forcemove/handle_click(mob/user, params, atom/A)
+	var/list/pa = params2list(params)
+	var/left_click = pa.Find("left")
+	var/right_click = pa.Find("right")
+
+	// Selecting the atom to move
+	if(right_click)
+		if(!ismovable(A) || iseffect(A))
+			to_chat(user, "<span class='danger'>You cannot move effects, turfs or areas.</span>")
+			return
+
+		// If we had something previously selected, handle its signal and overlay
+		if(selected_atom)
+			UnregisterSignal(selected_atom, COMSIG_PARENT_QDELETING)
+
+		if(selected_overlay)
+			remove_selected_overlay(user)
```

Comment:
This should happen the same time all the time right? So even when the item gets deleted it should also "unselect"  the overlay so it goes away. Basically the overlay shouldn't exist if the atom doesn't exist anymore

Reply:
There is for a very edge case where the atom gets destroyed after selecting it. In that case, the signal attached to the atom will make `selected_atom` null, but the `selected_overlay` stays as is (as I cannot handle `user.client.images` via the destroy signal).